matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/SessionWorkerParams.kt

```diff
@@ -27,5 +27,9 @@ internal interface SessionWorkerParams {
      * Null when no error occurs. When chaining Workers, first step is to check that there is no lastFailureMessage from the previous workers
      * If it is the case, the worker should just transmit the error and shouldn't do anything else.
      */
-    val lastFailureMessage: String?
+    var lastFailureMessage: String?
+}
+
+internal fun SessionWorkerParams.consumeFailure(): String? {
+    return lastFailureMessage.also { lastFailureMessage = null }
```

Comment:
Maybe we could avoid making `lastFailureMessage` a `var` by using the `data.copy` fun. It could be painful since we have an interface here, so every impl would have to do the copy.

Reply:
I was trying to avoid that as we have quite a few worker params, happy to make the change if preferred!

Reply:
Let's keep the `var` like this for now. Thanks!